test/e2e/serial/tests/resource_accounting.go

```diff
@@ -288,65 +294,108 @@ var _ = Describe("[serial][disruptive][scheduler] numaresources workload resourc
 			var ok bool
 			targetNodeName, ok = nrtCandidateNames.PopAny()
 			Expect(ok).To(BeTrue(), "cannot select a node among %#v", nrtCandidateNames.List())
-			By(fmt.Sprintf("selecting node to schedule the pod: %q", targetNodeName))
+			By(fmt.Sprintf("selecting node to schedule the test pod: %q", targetNodeName))
 
 			var err error
 			targetNodeNRTInitial, err = e2enrt.FindFromList(nrtCandidates, targetNodeName)
 			Expect(err).NotTo(HaveOccurred())
 
-			//get maximum zone CPU and Memory to be sure it wont fit on any zone
-			maxResources := corev1.ResourceList{
-				corev1.ResourceCPU:    maxResourceType(*targetNodeNRTInitial, corev1.ResourceCPU),
-				corev1.ResourceMemory: maxResourceType(*targetNodeNRTInitial, corev1.ResourceMemory),
-			}
+			//calculate base load on the target node
+			baseload, err := nodes.GetLoad(fxt.K8sClient, targetNodeName)
+			ExpectWithOffset(1, err).ToNot(HaveOccurred(), "missing node load info for %q", targetNodeName)
+			By(fmt.Sprintf("computed base load: %s", baseload))
 
-			// add a min ammount of resources just in case all the zones
-			// are equal so the pod wont fit on any of them
-			excessRes := corev1.ResourceList{
-				corev1.ResourceCPU:    resource.MustParse("100m"),
-				corev1.ResourceMemory: resource.MustParse("100M"),
+			//get maximum available node CPU and Memory
+			reqResources := corev1.ResourceList{
+				corev1.ResourceCPU:    availableResourceType(*targetNodeNRTInitial, corev1.ResourceCPU),
+				corev1.ResourceMemory: availableResourceType(*targetNodeNRTInitial, corev1.ResourceMemory),
 			}
-			reqResources := maxResources.DeepCopy()
-
-			reqCpu := reqResources[corev1.ResourceCPU]
-			reqCpu.Add(excessRes[corev1.ResourceCPU])
-			reqResources[corev1.ResourceCPU] = reqCpu
 
-			reqMem := reqResources[corev1.ResourceMemory]
-			reqMem.Add(excessRes[corev1.ResourceMemory])
-			reqResources[corev1.ResourceMemory] = reqMem
+			By("prepare the test's pod resources as maximum available resources on the target node with the baselaod deducted")
```

Comment:
what's the benefit of doing like this (which seems OK) over `e2enrt.SaturateNodeUntilLeft` ?

Reply:
good suggestion, I wasn't aware such a function already exists. Yet using `SaturateNodeUntilLeft` is not accurate with its current implementation, not particularly in this context but in general, because it returns the max resources to be padded on a node, but this actually unachievable because one container pod cannot be placed on 2 nodes (since we are in a single-numa-node mode). But I believe `SaturateNodeUntilLeft` can be adjusted by internally using `Saturate**Zone**UntilLeft`. note also that `SaturateNodeUntilLeft` is used by the deprecated makeNodePaddingPod

Reply:
now I wonder why lint didn't fail on leaving `SaturateNodeUntilLeft` while it is not used elsewhere